scim/resource_service_principal.go

```diff
@@ -161,7 +182,27 @@ func ResourceServicePrincipal() *schema.Resource {
 			})
 		},
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			return NewServicePrincipalsAPI(ctx, c).Delete(d.Id())
+			spAPI := NewServicePrincipalsAPI(ctx, c)
+			if c.AccountID != "" {
```

Comment:
logic is broken. this change means "user won't be deleted AT ALL if we are in the account context". 

i think we need so add some validation/explanation/documentation that deleting home dirs has no effect on the account level.